### DISCUSSION TITLE
File transfer after attaching the device to WSL

### DISCUSSION THREAD

**Discussion by kphahn** [dorssel/usbipd-win#141] [Q&A] [answered]
Hi,

I'm pretty new to all of this, so please excuse me if this might be trivial:

I used to mount my microcontroller as the d drive using the `mount` command and move my binary file onto it using `cp`. I did the debugging on windows side. I am happy that I can do the debugging with WSL now, but after attaching my microcontroller using `usbipd wsl attach`, I cannot mount it anymore. The message I get:

```
<4>WARNING: mount: waiting for virtio device...
<3>init: (1291) ERROR: MountPlan9WithRetry:285: mount drvfs on /mnt/d (cache=mmap,rw,msize=262144,trans=virtio,aname=drvfs;path=D:;symlinkroot=/mnt/) failed: 2
mount: No such file or directory
```

 This probably makes sense, but how would I now transfer my files without having to `detach` the device?

**Comment by dorssel** [comment]
Upon attach, the entire device is now "transferred" to WSL, so you would have to mount it there.

Some more info is required:
- What is the output of `lsusb`?
- Once attached, what is in `/var/log/syslog`? (you may have to install/start `rsyslog` first...)
- Do you see any extra `/dev/sdX` block devices after attach?

**Reply by kphahn** [reply]
`lsusb`:

```
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 005: ID 0483:374b STMicroelectronics ST-LINK/V2.1
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```




Had to figure out how to get the `syslog` running, but I hope this is what you were looking for:
```
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.440665] vhci_hcd: vhci_device speed not set
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.510766] usb 1-1: new full-speed USB device number 7 using vhci_hcd
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.590897] vhci_hcd: vhci_device speed not set
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.660695] usb 1-1: SetAddress Request (7) to port 0
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.697563] usb 1-1: New USB device found, idVendor=0483, idProduct=374b, bcdDevice= 1.00
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.697566] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.697567] usb 1-1: Product: STM32 STLink
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.697569] usb 1-1: Manufacturer: STMicroelectronics
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.697570] usb 1-1: SerialNumber: 066CFF515250898367112542
Dec  3 23:26:14 DESKTOP-TTFC8IO kernel: [ 7673.706173] cdc_acm 1-1:1.2: ttyACM0: USB ACM device
```



I don't see any new `/dev/sdX`. Only `/dev/ttyACM0` is new, which is where I can read data from with minicom. But I am still not sure how to put my file onto the device, as nothing happens when I try to copy it to `/dev/ACM0`.

**Comment by dorssel** [comment, marked as answer]
@kphahn 
That all looks normal. I guess for Linux you would have to use dedicated STLink tools to use the device. A quick google gave me `st-probe`, `st-flash`, `openocd`, etc. I am not familiar with STLink, but from those logs it appears that `usbipd-win` is working properly, i.e. no errors or anything.

**Reply by kphahn** [reply]
Okay, I think I now understand my actual problem, which is not `usbipd-win` related (like I said, it's all quite new to me).

Thank you for your help!
